solc/CommandLineInterface.cpp

```diff
@@ -716,8 +716,17 @@ bool CommandLineInterface::processInput()
 	if (m_args.count(g_argAllowPaths))
 	{
 		vector<string> paths;
-		for (string const& path: boost::split(paths, m_args[g_argAllowPaths].as<string>(), boost::is_any_of(",")))
-			m_allowedDirectories.push_back(boost::filesystem::path(path));
+		for (string const& path: boost::split(paths, m_args[g_argAllowPaths].as<string>(), boost::is_any_of(","))) {
+			auto filesystem_path = boost::filesystem::path(path);
+			// If the given path had a trailing slash, the Boost filesystem
+			// path will have it's last component set to '.'. This breaks
+			// path comparison in later parts of the code, so we need to strip
+			// it.
+			if (filesystem_path.filename() == ".") {
```

Comment:
The boost::path also has remove_trailing_separator() helper. Can you check if it works for this case.
Another alternative is canonical() http://www.boost.org/doc/libs/1_65_1/libs/filesystem/doc/reference.html#canonical but it might be too much.

Reply:
boost::path::canonical creates a canonical path based on the actual filesystem on which the code is running, including throwing an exception if the given path doesn't exist - we'd have to catch those and probably ignore them. However, if we do that, we would also be able to use relative paths in --allow-paths (currently this doesn't work).

WDYT? I'm for using canonical().

Reply:
How about using `remove_trailing_separator()` in this change and creating a new PR which fixes the other issue?

Reply:
I'm not sure if `canonical()` is not too big. E.g. it resolved symbolic links and this might be confusing for users that passed one path and will see different path in errors.